webapp/sources/rudder/rudder-web/src/main/elm/sources/Editor/ViewMethod.elm

```diff
@@ -125,39 +119,48 @@ accumulateValidationState validations base =
 accumulateErrorConstraint call constraints base =
-  accumulateValidationState (List.map  (checkConstraintOnParameter call) constraints ) base
+  accumulateValidationState [ (checkConstraintOnParameter call constraints) ] base
 
 checkConstraintOnParameter: CallParameter -> Constraint -> ValidationState MethodCallParamError
 checkConstraintOnParameter call constraint =
-  case constraint of
-    AllowEmpty          True -> ValidState
-    AllowEmpty          False -> if (isEmptyValue call.value) then InvalidState [ConstraintError { id = call.id, message = ("Parameter '"++call.id.value++"' is empty")}] else ValidState
-    AllowWhiteSpace     True -> ValidState
-    AllowWhiteSpace     False -> case Regex.fromString "(^\\s)|(\\s$)" of
-                               Nothing -> ValidState
-                               Just r -> if Regex.contains r (displayValue call.value) then InvalidState [ConstraintError { id = call.id, message = ( "Parameter '"++call.id.value++"' start or end with whitespace characters"  ) } ] else ValidState
-    MaxLength max -> if lengthValue call.value >= max then  InvalidState [ConstraintError  { id = call.id, message = ("Parameter '"++call.id.value++"' should be at most " ++ (String.fromInt max) ++ " long" ) } ]else ValidState
-    MinLength min -> if lengthValue call.value <= min then  InvalidState [ConstraintError { id = call.id, message = ("Parameter '"++call.id.value++"' should be at least " ++ (String.fromInt min) ++ " long") } ] else ValidState
-    MatchRegex r -> case Regex.fromString r of
-                      Nothing ->  ValidState
-                      Just regex -> if Regex.contains regex (displayValue call.value) then
-                                      ValidState
-                                    else
-                                       InvalidState [ConstraintError { id = call.id, message = ( "Parameter '" ++ call.id.value ++"' should match the following regexp: " ++ r  )} ]
-    NotMatchRegex r -> case Regex.fromString r of
-                      Nothing ->  ValidState
+  let
+    checkEmpty =  if not (constraint.allowEmpty  |> Maybe.withDefault False) && isEmptyValue call.value then [ConstraintError { id = call.id, message = ("Parameter '"++call.id.value++"' is empty")}] else []
+    checkWhiteSpace =
+      if constraint.allowWhiteSpace  |> Maybe.withDefault False then
+        []
+      else
+        case Regex.fromString "(^\\s)|(\\s$)" of
+          Nothing -> []
+          Just r -> if Regex.contains r (displayValue call.value) then [ConstraintError { id = call.id, message = ( "Parameter '"++call.id.value++"' start or end with whitespace characters"  ) } ] else []
+
+    checkMax = if lengthValue call.value >= (constraint.maxLength  |> Maybe.withDefault 16384) then  [ConstraintError  { id = call.id, message = ("Parameter '"++call.id.value++"' should be at most " ++ (String.fromInt (constraint.maxLength |> Maybe.withDefault 16384) ) ++ " long" ) } ]else []
```

Comment:
The default value `16384` should be defined somewhere globally and documented, in DataTypes likelly (it should not be in the middle of a view method)
I belived it should be the same for the name of the constraints, and their respective defaults (and perhaps the standard error message).